ivy.el

```diff
@@ -3117,11 +3135,219 @@ This function serves as a fallback when nothing else is available."
+                          (ivy--fuzzy-match-score
+                           (1- i) (1- j) t pat pat-role pat-set text text-role))
+                       (c1 i j)))
+              (cl-decf i)
+              (push (1- j) plan))
+          ;; Trace back to pat[i] and text[j-1].
+          (setq is-match
+                (= (+ (c1 i (1- j))
+                      (ivy--fuzzy-miss-score (1- j) t text-role))
+                   (c0 i j))))
+        (cl-decf j))
+      (nreverse plan))))
+
+(defun ivy--fuzzy-sort (pat cands)
+  "Sort according to closeness to string PAT the string list CANDS."
+  (let ((regex (mapconcat
+                (lambda (x)
+                  (format "\\(%s\\)" (regexp-quote (string x))))
+                (delete ?\  (string-to-list pat))
+                ".*?")))
+    (setq cands (cl-delete-if-not (apply-partially #'string-match regex) cands)))
+  (let* ((m (length pat))
+         (n (apply #'max 0 (mapcar #'length cands)))
+         pat-set
+         text-set
+         ;; Vectors of roles for each position of pat and text.
+         (pat-role (make-vector m nil))
+         (text-role (make-vector n nil))
+         ;; match and miss are 2D DP vectors.
+         (miss (make-vector (1+ m) nil))
+         (match (make-vector (1+ m) nil))
+         cands-to-sort
+         cands-with-score
+         plan)
+    (setq pat-set (ivy--fuzzy-get-role pat pat-role))
+    (setq m 0)
+    (dotimes (i (length pat))
+      (when (/= (aref pat i) ?\ )
+        (aset pat m (aref pat i))
+        (aset pat-role m (aref pat-role i))
+        (cl-incf m)))
+    (setq pat (substring-no-properties pat 0 (min m ivy-fuzzy-pattern-length-limit)))
+
+    (dotimes (i (1+ m))
+      (setf (aref miss i) (make-vector (1+ n) 0)
+            (aref match i) (make-vector (1+ n) 0)))
+
+    ;; Partition the candidates into sorted and unsorted groups.
+    (dotimes (_i (min n ivy-fuzzy-limit))
+      (push (pop cands) cands-to-sort))
+
+    ;; For each candidate, compute the score.
+    (dolist (text cands-to-sort)
+      (if (> (length text) ivy-fuzzy-candidate-length-limit)
+          (setq text (substring text 0 ivy-fuzzy-candidate-length-limit))
+        (setq text (copy-sequence text)))
+      (setq text-set (ivy--fuzzy-get-role text text-role))
+      (pcase-let ((`(,score . ,best-pos)
+                   (ivy--fuzzy-score pat pat-role pat-set
+                                     text text-role text-set miss match)))
+        ;; Discard awful candidates.
+        (when best-pos
+          (dolist (pos (ivy--fuzzy-plan pat pat-role pat-set
+                                        text text-role text-set miss match best-pos))
+            (ivy-add-face-text-property pos (1+ pos) (cadr ivy-minibuffer-faces) text))
+          (push (cons score text) cands-with-score))))
+    (append (mapcar #'cdr (cl-sort cands-with-score #'> :key #'car))
+            cands)))
+
+(define-obsolete-variable-alias 'ivy-flx-limit 'ivy-fuzzy-limit)
```

Comment:
This should be near the variable in question, specifically before it (see the docstring of `define-obsolete-variable-alias`). The `WHEN` argument should also be supplied before merging.